context/app/static/js/components/workspaces/AdvancedConfigOptions/AdvancedConfigOptions.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useCallback, useMemo } from 'react';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import Button from '@mui/material/Button';
+import Slider from '@mui/material/Slider';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/system/Stack';
+import ArrowDropDownRounded from '@mui/icons-material/ArrowDropDownRounded';
+
+import { ControllerRenderProps, FieldValues, Path, useController, UseControllerProps } from 'react-hook-form';
+import InfoTooltipIcon from 'js/shared-styles/icons/TooltipIcon';
+import { SpacedSectionButtonRow } from 'js/shared-styles/sections/SectionButtonRow';
+import { StyledAccordion, StyledHeader, StyledSubtitle, StyledSwitch, StyledSwitchLabel } from './style';
+import {
+  DEFAULT_GPU_ENABLED,
+  DEFAULT_MEMORY_MB,
+  DEFAULT_NUM_CPUS,
+  DEFAULT_TIME_LIMIT_MINUTES,
+  MAX_MEMORY_MB,
+  MAX_NUM_CPUS,
+  MAX_TIME_LIMIT_MINUTES,
+  MIN_MEMORY_MB,
+  MIN_NUM_CPUS,
+  MIN_TIME_LIMIT_MINUTES,
+} from '../constants';
+
+interface ConfigSliderProps<FormType extends FieldValues> {
+  field: ControllerRenderProps<FormType, Path<FormType>>;
+  id: string;
+  label: string;
+  tooltip: string;
+  min: number;
+  max: number;
+  conversionFactor?: number;
+  markInterval?: number;
+}
+
+function ConfigSlider<FormType extends FieldValues>({
+  field,
+  id,
+  label,
+  tooltip,
+  min,
+  max,
+  conversionFactor = 1,
+  markInterval = 1,
+}: ConfigSliderProps<FormType>) {
+  const convert = (value: number) => value / conversionFactor;
+  const unconvert = (value: number) => value * conversionFactor;
+
+  const convertedMin = convert(min);
+  const convertedMax = convert(max);
+
+  const marks = useMemo(() => {
```

Comment:
This isn't a particularly expensive computation so I don't believe it needs to be memoized. Also prefer to use `map` instead of the `for` loop.